Add Checklist for Codereview in Developer Guide (#6179)
gammapy/gammapy: docs/development/dev_howto.rst

where they have fetched the `gammapy-data <https://github.com/gammapy/gammapy-data>`__ GitHub repository,
so they can push and pull eventual modification of its content.

> .. _skip_actions_pr:

Reviewer: ```suggestion
.. _skip-actions-pr:
```

but that the core developer adds a releases notes entry after right after having
merged a pull request (you can add ``[skip ci]`` on this commit).

How to review a pull request?
-----------------------------

The checklist below outlines key steps to follow when reviewing a PR.

1. Ensure that a milestone and labels are correctly set for the PR.
2. Check that the code does what it is supposed to do.
3. Check that the CI workflow passes without failures.
4. Check that the general conventions are fulfilled, :ref:`see here <general-conventions>`.
5. Check that the code can be understood by reading it.
6. Confirm all docstrings are correct and formatted correctly, :ref:`see here <docstring-formatting>`.
7. Check that the PR links to associated issue(s), if any.

> 8. Check that all commits are signed.

Reviewer: Good question, I am not sure of the procedure there. @registerrier ?

> 9. Add a fragment, if necessary, for towncrier.

Author: ```suggestion
9. Check that a fragment was added, if necessary, for towncrier.
```
Would this also be a check?

Reviewer: No worries! 
There is a ref in my towncrier PR called `release-notes`, so once that is merged this PR can be followed and will render correctly. 

Not sure I understand the the points about matplotlib?

Author: Thank you, alright! I tried to put it in with` :ref:` `towncrier <release-notes>` and in the moment it links for me to https://matplotlib.org/stable/users/release_notes.html#release-notes`, could it be that there is a another reference called `release-notes` somewhere?

but that the core developer adds a releases notes entry after right after having
merged a pull request (you can add ``[skip ci]`` on this commit).

> How to review a pull request?

Reviewer: ```suggestion
How to review a pull request
```

If the path of the output file is not provided, the script will be written in the same folder as the notebook and with
the same name.

> .. _skip_actions_local_fork:

Reviewer: ```suggestion
.. _skip-actions-local-fork:
```

Skip GitHub Actions on local fork
+++++++++++++++++++++++++++++++++
If not explicitly needed, it can be convenient to skip the GitHub Actions for pushes onto local forks and only perform them on the upstream, once a pull request is opened.
This way computation power is saved and automatic commits by the Actions are avoided in the forks commit-history. An easy way to achieve this, is to deactivate the
GitHub Actions completely for the fork, following the GitHub documentation, `see here. <https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#managing-github-actions-permissions-for-your-repository>`__

> See also the section on making a pull request which skips GitHub Actions, :ref:`here <skip_actions_pr>`.

Reviewer: ```suggestion
See also the section on making a pull request which skips GitHub Actions, :ref:`here <skip-actions-pr>`.
```

If not explicitly needed, it can be convenient to skip the GitHub Actions for pushes onto local forks and only perform them on the upstream, once a pull request is opened.
This way computation power is saved and automatic commits by the Actions are avoided in the forks commit-history. An easy way to achieve this, is to deactivate the
GitHub Actions completely for the fork, following the GitHub documentation, `see here. <https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#managing-github-actions-permissions-for-your-repository>`__

> See also section: Making a pull request which skips GitHub Actions, above.

Reviewer: I would add a similar ref like you did for the others. 
And also for the other section `Making a pull request which skips GitHub Actions`.

Author: Thank you very much for your suggestion, I applied it and did the missing changes in the other file!
